src/controllers/scripting/legacy/controllerscriptenginelegacy.cpp

```diff
@@ -630,6 +511,8 @@ void ControllerScriptEngineLegacy::handleScreenFrame(
         return;
     };
 
+    auto& pScreen = m_rootItems.at(screenInfo.identifier);
```

Comment:
consider using `find` to handle the case where its not in the map, otherwise it'll throw.

Reply:
Line 509, there is already an assertin for the string to be in the map

Reply:
Ah right. Well the `contains`+`at` pattern is suboptimal because you do two lookups instead of one. It'd be nice if you could switch these around to only use one `find`.

Reply:
Makes sense. I have rework it to use a single lookup. Hopefully this works as you had it in mind.

Reply:
yup, pretty much exactly. thanks.